src/Network/XKoreProxy.pm

```diff
@@ -79,15 +81,14 @@ sub new {
 		$self->{packetPending} = '';
 		$clientBuffer = '';
 	}
-
+	
+	$packetParser = Network::Receive->create($self, $masterServer->{serverType});
```

Comment:
This should be done in eval { } so we can catch Exception::Class::Base (the same way it's done in sub checkServer)